File: ratio_backend/domain/company.py
"""Company domain types used by repositories and application services."""

from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum


class CompanyVisibility(str, Enum):
    """Public visibility states for a tracked company."""

    DRAFT_ONLY = "draft_only"
    PUBLISHED = "published"
    HIDDEN = "hidden"


@dataclass(slots=True)
class CompanyRecord:
    """Canonical company metadata shared by public and admin surfaces."""

    ticker: str
    company_name: str
    display_name: str | None = None
    sector: str | None = None
    industry: str | None = None
    description: str | None = None
    visibility: CompanyVisibility = CompanyVisibility.DRAFT_ONLY
    is_tracked: bool = True
    is_published: bool = False
    display_order: int = 0
    created_at: datetime | None = None
    updated_at: datetime | None = None
    id: int | None = None
    metadata: dict[str, str] = field(default_factory=dict)

File: crates/ratio-domain/src/company.rs
//! Company entities shared by admin and public backend layers.

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum CompanyVisibility {
    DraftOnly,
    Published,
    Hidden,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Company {
    pub id: Option<i64>,
    pub ticker: String,
    pub company_name: String,
    pub display_name: Option<String>,
    pub sector: Option<String>,
    pub industry: Option<String>,
    pub description: Option<String>,
    pub visibility: CompanyVisibility,
    pub is_tracked: bool,
    pub is_published: bool,
    pub display_order: i32,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
}
